test-structure.md

```diff
@@ -0,0 +1,89 @@
+There is a standard template we use for our tests in Java and Swift. This helps us keep our test files organized so we can
+easily keep track of the outputs and flows we are testing.
+
+## Java
+```java
+// { List of imports }
+
+public final class MyViewModelTest extends KSRobolectricTestCase {
+  private MyViewModel.ViewModel vm;
+  
+  // { Alphabetized declaration of output test subscribers }
+  
+  private final TestSubscriber<Boolean> projectNameTextViewHidden = new TestSubscriber<>();
+  private final TestSubscriber<String> projectNameTextViewText = new TestSubscriber<>();
+  
+  // { Private set up environment method at the top }
+  
+  private void setUpEnvironment(final @NonNull Environment environment) {
```

Comment:
nice! glad we are documenting this _and_ hopefully we can find a way in the future to reduce some of the boilerplate.